### PR TITLE
Updates - postgres 9.4 breaking change

### PR DESCRIPTION
**Pull request by jsa34**
-Various updates and security fixes prior to any dev work

### CODE REVIEW

**Comment by jsa34** [comment]
Now finished.

To note: 

- migrated json cols to jsonb due to: https://github.com/bookshelf/bookshelf/issues/1941
- I believe the above is why it doesn't work anymore on Postgres 9.4, but is fine for versions above.
- Also noticing a couple of instances of the following stacktrace in unit tests - appears to be unhandled error that may be important, but can't see why it is unhappy.  Unit tests pass and the table referenced APPEARS to be ok ... ?

**Unhandled rejection error: relation "sessions" does not exist
    at Connection.parseE (C:\Users\jsa34\WebstormProjects\sidekick\node_modules\pg\lib\connection.js:604:11)
    at Connection.parseMessage (C:\Users\jsa34\WebstormProjects\sidekick\node_modules\pg\lib\connection.js:401:19)
    at Socket.<anonymous> (C:\Users\jsa34\WebstormProjects\sidekick\node_modules\pg\lib\connection.js:121:22)
    at Socket.emit (events.js:198:13)
    at addChunk (_stream_readable.js:288:12)
    at readableAddChunk (_stream_readable.js:269:11)
    at Socket.Readable.push (_stream_readable.js:224:10)
    at TCP.onStreamRead [as onread] (internal/stream_base_commons.js:94:17)
**

File: model/setting.js [hunk 1]
```diff
@@ -87,7 +87,8 @@ function initSettingModel(dashboard) {
 				});
 				await setting.save(null, {
 					method: 'insert'
-				});
+				}
+				);
```
Comment (joeyciechanowicz) [inline model/setting.js]: Super petty, but can you put this on the line above? 🙇

File: README.md [hunk 2]
```diff
@@ -28,7 +28,7 @@ This is a new project based around Pa11y, code-named "Sidekick". Pa11y Sidekick
 
 ## Requirements
 
-This application requires [Node.js] 4+ and [PostgreSQL] 9.4+.
+This application requires [Node.js] 4+ and [PostgreSQL] 9.5+.
```
Comment (joeyciechanowicz) [inline README.md]: We dropped Node 4 on pa11y a while ago, it's now Node 8+
```suggestion
This application requires [Node.js] 8+ and [PostgreSQL] 9.5+.
```

Just checking the travis.yml as well, and that's Node 8 anyway!! 🎉

File: lib/dashboard.js [hunk 3]
```diff
@@ -103,7 +103,7 @@ class Dashboard {
 	/**
 	 * Create a Bookshelf database connection.
 	 * @param {String} connectionString - A PostgreSQL connection string.
-	 * @returns {Bookshelf} A Bookshelf instance.
+	 * @returns {{VERSION: *}|{VERSION: *}} A Bookshelf instance.
```
Comment (joeyciechanowicz) [inline lib/dashboard.js]: Is this a refactor gone wrong? Doesn't seem quite like a valid type...

**Review by joeyciechanowicz — CHANGES_REQUESTED** [review]
Couple of minor changes, but fantastic work getting this back and running!

File: model/setting.js (hunk 1 above)
Comment (jsa34) [reply model/setting.js]: Apologies - well spotted!

File: lib/dashboard.js (hunk 3 above)
Comment (jsa34) [reply lib/dashboard.js]: Gosh - yup... Not sure how that happened....?  Sounds like WebStorm just had a field day doing something silently - I don't remember changing this!

**Review by joeyciechanowicz — APPROVED** [review]

**Pull request merged**